pycaret/loggers/comet_logger.py

```diff
@@ -0,0 +1,69 @@
+from pycaret.loggers.base_logger import BaseLogger
+from pathlib import Path
+import uuid
+from copy import deepcopy
+import joblib
+
+try:
+    import comet_ml
+except ImportError:
+    comet_ml = None
+
+class CometLogger(BaseLogger):
+    def __init__(self) -> None:
+        if comet_ml is None:
+            raise ImportError(
+                "CometLogger requires Comet. Install using `pip install comet_ml`"
+            )
+        super().__init__()
+        self.run = None
+    
+    def init_experiment(self, exp_name_log, full_name=None, **kwargs):
+        self.run = comet_ml.Experiment(project_name=exp_name_log, **kwargs)
+        self.run.set_name(full_name)
+        self.run.log_other('Created from', 'pycaret')
+        return self.run
+    
+    def log_params(self, params, model_name=None):
+        self.run.log_parameters(params, prefix=model_name)
+    
+    def set_tags(self, source, experiment_custom_tags, runtime):
+        tags = [source, runtime]
+        self.run.add_tags(tags)
+        self.run.log_others(experiment_custom_tags)
+
+    def log_sklearn_pipeline(self, experiment, prep_pipe, model, path=None):
+        
```

Comment:
to remove explicit dependency on `joblib` it's possible to add a method to set a "model_dumper" property on this class

```python
def __init__(self, pipline_dumper):
    self._pipeline_dumper = pipeline_dumper # user can pass joblib.dump or any other suitable function 
```


the price to pay is that it's not part of the standard API